src/main/java/org/madhuri/app/message/dao/ChannelUserDAO.java

```diff
@@ -0,0 +1,32 @@
+
+package org.madhuri.app.message.dao;
+
+import org.madhuri.app.message.model.Channel;
+import org.madhuri.app.message.model.User;
+
+public class ChannelUserDAO {
+	
+private ChannelDAO channelDAO = new ChannelDAO();
+    
+    public User addUserToChannel(User user, Channel channel) {
+    	try {
+            if (user == null || channel == null) {
+                throw new IllegalArgumentException("User and channel must not be null");
+            }
+
+            channel.addUser(user);
+            channelDAO.updateChannel(channel);
```

Comment:
why are you updating channel here ?

cannot we insert data into channel user table directly from here?
All we need to save here is user_id & channel_id in channel user table.

Reply:
just like you save user or channel to table, why cannot you save ChannelUser directly here ?
You already created a ChannelUser model

Reply:
1. create a ChannelUser object adding user_id & channel_id
2. session.Save()